components/forms/StageForm.tsx

```diff
@@ -0,0 +1,92 @@
+import { Button, DatePicker, Form, Input, Select } from 'antd';
+import { stageTypeToDisplayStringMap } from '../../utils/applicationStage/applicationStageUtils';
+import AddEmojiIcon from '../icons/AddEmojiIcon';
+import { Dispatch, MouseEvent, SetStateAction, useEffect, useState } from 'react';
+import dynamic from 'next/dynamic';
+import { IEmojiData } from 'emoji-picker-react';
+import { Nullable } from '../../types/utils';
+import { ApplicationStageFormData } from '../../types/applicationStage';
+
+interface Props {
+  initialValues: ApplicationStageFormData;
+  isSubmitting: boolean;
+  setIsSubmitting: Dispatch<SetStateAction<boolean>>;
+  setStageFormData: Dispatch<SetStateAction<Nullable<ApplicationStageFormData>>>;
+}
+
+function StageForm({ initialValues, isSubmitting, setIsSubmitting, setStageFormData }: Props) {
+  const [form] = Form.useForm();
+
+  const [shouldShowEmojiPicker, setShouldShowEmojiPicker] = useState<boolean>(false);
+  const [selectedEmojiUnicode, setSelectedEmojiUnicode] = useState<Nullable<string>>(null);
+
+  const DynamicPicker = dynamic(() => import('emoji-picker-react'));
+
+  const onSelectEmoji = (event: MouseEvent, emojiObject: IEmojiData) => {
+    setSelectedEmojiUnicode(emojiObject.unified);
+    setShouldShowEmojiPicker(false);
+  };
+
+  useEffect(() => {
+    form.resetFields();
+
+    if (initialValues.emojiUnicodeHex !== undefined) {
+      setSelectedEmojiUnicode(initialValues.emojiUnicodeHex);
+    }
+  }, [initialValues]);
+
+  useEffect(() => {
+    if (isSubmitting) {
+      form
+        .validateFields()
+        .then(() => setStageFormData({ ...form.getFieldsValue(), emojiUnicodeHex: selectedEmojiUnicode }))
+        .catch(() => setIsSubmitting(false));
+    }
+  }, [isSubmitting]);
+
+  return (
+    <Form form={form} initialValues={initialValues}>
```

Comment:
![image](https://user-images.githubusercontent.com/75519136/190921867-687f57e6-4e77-46f7-83a8-69f8a536dbce.png)

UI nit but can we align the start of the fields?
```suggestion
    <Form form={form} initialValues={initialValues} labelCol={{ span: 8 }} wrapperCol={{ span: 16}}>
```

Something like that should work, though the `span` values should be adjusted.